app/features/calculator.py

```diff
@@ -0,0 +1,892 @@
+    )
+    async def calc_sqrt(self, ctx: interactions.SlashContext, number: str) -> None:
+        """Calculate the square root of a number."""
+        try:
+            result = evaluate_expression(number)
+            await ctx.send(f"The square root of {number} is: {result}")
+        except (CalculationError, ValueError, TypeError) as e:
+            await ctx.send(f"An error occurred: {e!s}", ephemeral=True)
+
+    @calc.subcommand(sub_cmd_name="root", sub_cmd_description="Calculate the nth root of a number")
+    @interactions.slash_option(
+        name="x",
+        description="The number to find the root of",
+        required=True,
+        opt_type=interactions.OptionType.STRING,
+    )
+    @interactions.slash_option(
+        name="n",
+        description="The root exponent",
+        required=True,
+        opt_type=interactions.OptionType.STRING,
+    )
+    async def calc_root(self, ctx: interactions.SlashContext, x: str, n: str) -> None:
+        """Calculate the nth root of a number."""
+        try:
+            expression = f"root({x}, {n})"
+            result = evaluate_expression(expression)
+            await ctx.send(f"The {n}th root of {x} is: {result}")
+        except (CalculationError, ValueError, TypeError) as e:
+            await ctx.send(f"An error occurred: {e!s}", ephemeral=True)
+
+    @calc.subcommand(sub_cmd_name="ln", sub_cmd_description="Calculate the natural logarithm")
+    @interactions.slash_option(
+        name="x",
+        description="The number to calculate the natural logarithm of",
+        required=True,
+        opt_type=interactions.OptionType.STRING,
+    )
+    async def calc_ln(self, ctx: interactions.SlashContext, x: str) -> None:
+        """Calculate the natural logarithm of a number."""
+        try:
+            expression = f"ln({x})"
+            result = evaluate_expression(expression)
+            await ctx.send(f"The natural logarithm of {x} is: {result}")
+        except (CalculationError, ValueError, TypeError) as e:
+            await ctx.send(f"An error occurred: {e!s}", ephemeral=True)
+
+    @calc.subcommand(sub_cmd_name="log", sub_cmd_description="Calculate the logarithm with a specified base")
+    @interactions.slash_option(
+        name="x",
+        description="The number to calculate the logarithm of",
+        required=True,
+        opt_type=interactions.OptionType.STRING,
+    )
+    @interactions.slash_option(
+        name="base",
+        description="The base of the logarithm (default is 10)",
+        required=False,
+        opt_type=interactions.OptionType.STRING,
+    )
+    async def calc_log(self, ctx: interactions.SlashContext, x: str, base: str = "10") -> None:
+        """Calculate the logarithm of a number with a specified base."""
+        try:
+            expression = f"log({x}, {base})"
+            result = evaluate_expression(expression)
+            await ctx.send(f"The logarithm of {x} with base {base} is: {result}")
```

Comment:
The log function seems to always result in a Syntax error.
`/calc log 16 2` as well as `/calculate log(16, 2)`